testsuite/features/step_definitions/vm_steps.rb

```diff
@@ -0,0 +1,432 @@
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  when 'sles'
+    name = 'sles-disk-image-template.qcow2'
+    net = 'salt-sles'
+    os = 'sle15sp4'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  when 'rhlike'
+    name = 'rhlike-disk-image-template.qcow2'
+    net = 'salt-rhlike'
+    os = 'rocky8'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  when 'deblike'
+    name = 'deblike-disk-image-template.qcow2'
+    net = 'salt-deblike'
+    os = 'ubuntu2204'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  else
+    name = 'empty'
+  end
+  raise "/var/testsuite-data/#{name} not found" unless file_exists?(node, "/var/testsuite-data/#{name}")
+
+  node.run("cp /var/testsuite-data/#{name} #{disk_path}")
+
+  # Actually define the VM, but don't start it
+  raise 'not found: virt-install' unless file_exists?(node, '/usr/bin/virt-install')
+
+  # with cloud init image
+  if cloudinit == 'with'
+    raise "#{cloudinit_image} not found" unless file_exists?(node, cloudinit_image)
+
+    node.run("cp #{cloudinit_image} #{cloudinit_path}")
+    node.run(
+      "virt-install --name #{vm_name} --memory #{memory} --vcpus 1 --disk path=#{disk_path},bus=virtio " \
+      "--network network=#{net} --graphics vnc,listen=0.0.0.0 " \
+      "--serial file,path=/tmp/#{vm_name}.console.log " \
+      "--import --hvm --noautoconsole --noreboot --osinfo #{os} " \
+      "--disk path=#{cloudinit_image},device=cdrom,bus=sata"
+    )
+  else
+    node.run(
+      "virt-install --name #{vm_name} --memory #{memory} --vcpus 1 --disk path=#{disk_path},bus=virtio " \
+      "--network network=#{net} --graphics vnc,listen=0.0.0.0 " \
+      "--serial file,path=/tmp/#{vm_name}.console.log " \
+      "--import --hvm --noautoconsole --noreboot --osinfo #{os} "
+    )
+  end
+end
+
+When(/^I stop the virtual machine named "([^"]*)" on "([^"]*)"$/) do |vm_name, host|
+  node = get_target(host)
+  _running, code = node.run("virsh list --name | grep #{vm_name}", check_errors: false)
+  next unless code.zero?
+  raise 'not found: virsh' unless file_exists?(node, '/usr/bin/virsh')
+
+  node.run("virsh destroy #{vm_name}")
+end
+
+When(/^I delete the virtual machine named "([^"]*)" on "([^"]*)"$/) do |vm_name, host|
+  node = get_target(host)
+  raise 'not found: virsh' unless file_exists?(node, '/usr/bin/virsh')
+
+  step %(I stop the virtual machine named "#{vm_name}" on "#{host}")
+  node.run("virsh undefine --nvram #{vm_name}")
+end
+
+When(/^I create ([^ ]*) virtual network on "([^"]*)"$/) do |net_name, host|
+  node = get_target(host)
+
+  networks = {
+    'test-net0' => { 'bridge' => 'virbr0', 'subnet' => 124 },
+    'test-net1' => { 'bridge' => 'virbr1', 'subnet' => 126 },
+    'salt-sles' => { 'bridge' => 'br0' },
+    'salt-leap' => { 'bridge' => 'br0' },
+    'salt-rhlike' => { 'bridge' => 'br0' },
+    'salt-deblike' => { 'bridge' => 'br0' }
+  }
```

Comment:
I would move this constant to our constants.rb file

Reply:
This is also something for another future PR. I just moved this step.